src/lib/y2storage/proposal/lvm_creator.rb

```diff
@@ -102,8 +102,11 @@ def create_volume_group(planned_vg, devicegraph)
       # @param devicegraph [Devicegraph] to fetch the partitions
       def assign_physical_volumes(volume_group, part_names, devicegraph)
         partitions = part_names.map { |n| BlkDevice.find_by_name(devicegraph, n) }
+        used_devices = volume_group.lvm_pvs.map(&:blk_device)
         partitions.each do |partition|
+          next if used_devices.include?(partition) # already a PV for this VG
```

Comment:
So here we only check for partitions that are directly a PV (no encryption) but in the next line, we seem to cope more or less transparently with possible encryption layer. My arachnid sense screams.

Reply:
Wouldn't it be more consistent using?
```ruby
used_devices = volume_group.lvm_pvs.map(&:plain_blk_device)
```

Reply:
Another alternative would be to not have `used_devices` and do this check instead (I don't have a preference).
```ruby
next if partition.lvm_pv && partition.lvm_pv.lvm_vg == volume_group
```